backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+            # Return zero vector as fallback
+            return [0.0] * EMBEDDING_DIMENSION
+    
+    async def _store_chunks(self, material_id: int, chunk_data: List[Dict]) -> List[Dict]:
+        """Store chunks in the database"""
+        db = SessionLocal()
+        stored_chunks = []
+        
+        try:
+            for chunk_dict in chunk_data:
+                # Create content hash for deduplication
+                content_hash = hashlib.md5(chunk_dict['content'].encode()).hexdigest()
```

Ruff (0.14.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)